core/ustring.cpp

```diff
@@ -830,6 +830,61 @@ Vector<String> String::split_spaces() const {
 	return ret;
 }
 
+Vector<String> String::split_lines(bool p_allow_empty, int p_maxsplit) const {
+	Vector<String> ret;
+
+	// empty string guard
+	if (length() <= 0) {
+		ret.push_back(*this);
+		return ret;
+	}
+
+	int from = 0;
+	CharType curr_char;
+	CharType next_char = get(0);
+
+	for (int i = 0; i < length(); i++) {
+		curr_char = next_char;
+		next_char = get(i + 1); // will get the null terminator when i == length() - 1
+
+		if ((!p_allow_empty) && (i == from) && (curr_char == '\n' || curr_char == '\r')) { // line is empty and needs to be skipped
+			do { // speed past all newline chars
+				i++;
+				next_char = get(i);
+			} while (next_char == '\n' || next_char == '\r');
+			from = i;
+			continue;
+		}
+
+		switch (curr_char) {
+			case '\n':
+				ret.push_back(substr(from, i - from));
+				// could check for \n\r here, but ignoring for consistency with other engine functions
+				from = i + 1;
+				next_char = get(from);
+				break;
+			case '\r':
+				ret.push_back(substr(from, i - from));
+				if (next_char == '\n') { // skip \r\n as pair
+					i++;
+				}
+				from = i + 1;
+				next_char = get(from);
+				break;
+		}
+
+		// just pushed maximum number of lines
+		if (p_maxsplit > 0 && ret.size() >= p_maxsplit) {
+			break;
+		}
+	}
+
+	// no more newlines found, push the last line
+	ret.push_back(substr(from, -1)); // substr treats -1 as "the rest"
```

Comment:
```suggestion
	ret.push_back(substr(from)); // substr treats -1 as "the rest"
```
-1 is the default argument.

Reply:
The comment should be removed too if `-1` is removed, as it's not intelligible without it.